packages/@lwc/template-compiler/src/codegen/codegen.ts

```diff
@@ -275,7 +275,7 @@ export default class CodeGen {
     /**
      * Generates a factory function that inturn generates child vnodes for scoped slot content.
      */
-    getScopedSlotFactory(callback: t.FunctionExpression, slotName: t.SimpleLiteral) {
+    getScopedSlotFactory(callback: t.FunctionExpression, slotName: t.Expression | t.SimpleLiteral) {
```

Comment:
This type change hasn't been reflected in the following places in engine-core
1. `function ssf(slotName: string` @ `packages/@lwc/engine-core/src/framework/api.ts`
2. `interface VScopedSlotFragment`.`slotName` @ `packages/@lwc/engine-core/src/framework/vnodes.ts`